AirCasting/SearchAndFollow/SearchAndFollowMap/SearchAndFollowMap.swift

```diff
@@ -117,8 +131,39 @@ struct SearchAndFollowMap: UIViewRepresentable {
                 mapView.camera = mapCamera
                 
             }
-            parent.startingPoint = CLLocationCoordinate2D(latitude: mapView.camera.target.latitude, longitude: mapView.camera.target.longitude)
-            dotsPositionShouldChange(for: mapView)
+            if parent.startingPointChanged {
+                parent.startingPoint = CLLocationCoordinate2D(latitude: mapView.camera.target.latitude, longitude: mapView.camera.target.longitude)
+                startingPointHolder = parent.startingPoint
+                startPositionShouldChange(for: mapView)
+                DispatchQueue.main.async {
+                    self.parent.startingPointChanged = false
+                }
+            } else {
```

Comment:
the `else` clause is not needed since there is no code that follows:

```Swift
if parent.startingPointChanged {
     parent.startingPoint = CLLocationCoordinate2D(latitude: mapView.camera.target.latitude, longitude: mapView.camera.target.longitude)
     startingPointHolder = parent.startingPoint
     startPositionShouldChange(for: mapView)
     DispatchQueue.main.async {
         self.parent.startingPointChanged = false
     }
}
parent.startingPoint = CLLocationCoordinate2D(latitude: mapView.camera.target.latitude, longitude: mapView.camera.target.longitude)
startingPointHolder = parent.startingPoint
dotsPositionShouldChange(for: mapView)
```

Reply:
I think that having this kind of condition we will execute the `{ else }` part every time the first one will be executed. We don't want that as long as they do differ here: `startPositionShouldChange(for: mapView)` `dotsPositionShouldChange(for: mapView)`. I changed it a bit though

Reply:
sorry, you're right! I forgot to add a `return` statement into the `if` part.
Should look like this:
```Swift
if parent.startingPointChanged {
     parent.startingPoint = CLLocationCoordinate2D(latitude: mapView.camera.target.latitude, longitude: mapView.camera.target.longitude)
     startingPointHolder = parent.startingPoint
     startPositionShouldChange(for: mapView)
     DispatchQueue.main.async {
         self.parent.startingPointChanged = false
     }
     return
}
parent.startingPoint = CLLocationCoordinate2D(latitude: mapView.camera.target.latitude, longitude: mapView.camera.target.longitude)
startingPointHolder = parent.startingPoint
dotsPositionShouldChange(for: mapView)
```